fluentd/lib/fluentd-okd-config-generator/lib/fluentd-okd-config-generator/templates/store_elasticsearch.erb

```diff
@@ -0,0 +1,41 @@
+<store>
+    @type elasticsearch
+    @id <%= store_id(@target, @context) %>
+    host "<%= @target[:host] %>"
+    port "<%= @target[:port] %>"
+    scheme https
```

Comment:
So this store is only to be used with the internal cluster Elasticsearch, not an external customer supplied Elasticsearch.

Reply:
Based on the input in the comment [1] and the description of in the design document, I don't see where this stanza restricts you to an internal cluster.  Assuming we have:
* a resolveable destination
* proper certificates to connect to the endpoints

is there more needed to route externally?

[1] https://github.com/openshift/origin-aggregated-logging/pull/1526/files#diff-a14b23e80be15decfe6c74cb2499ac83R10

Reply:
Yes - there are some customers who
- want to use http not https (so add the scheme to the endpoint e.g. use `https://host:port`)
- want to use username/password auth
- want to use token auth (provide custom http headers)
- want other stuff too

so - what we have now is fine for the internal cluster Elasticsearch, and probably good enough for some fraction of external cluster Elasticsearch use cases

Reply:
> * want to use http not https (so add the scheme to the endpoint e.g. use `https://host:port`)
This can be inferred by the presence (or not) of secrets

> * want to use username/password auth
> * want to use token auth (provide custom http headers)
> * want other stuff too

I'm thinking of this purely from the perspective of the need to define something in a ClusterLogging object. I intentionally want to avoid providing customers with EVERY possible configuration as that would need to be surfaced to CL.  IMO, these are permutations we not desire to test and support; our team just doesn't have the coverage.  I believe we need to be extremely opinionated.

One case in point I was investigating yesterday is kafka.  The kafka plugin allows for quite a bit of config which is not common to any other plugins (e.g. topic, broker type).  I believe this is another place where we can establish a pattern like we always writing messages to a topic that is named the same as the source.

Reply:
ok - Then this config as you have it is fine because it solves our most immediate use cases